openssl: fix 1.0.0t, update 1.0.1h→1.0.1r, fix 1.0.2f

fbrosson opened the pull request.

fbrosson commented:
I'm not sure whether my changes in the REQUIRES section of 1.0.2f are required/desirable or not.
Please also check if my changes about the "compat" versions are all OK.
Don't hesitate to tell me if I missed something!
Regarding the upgrade from 1.0.1h to 1.0.1t, I'll give it a try in a few days unless someone else takes it:)

pulkomandy commented:
The version of the libraries must match the sonames. You can check the package content, if the lib is named libcrypto.so.1.0.0 then the version must be 1.0.0 (and if there is a symlink named libcrypto.so.1.0, then the compat should be ">= 1.0").
This is unrelated to the portversion, even if it matches for projects using a sane versionning scheme, which I think OpenSSL isn't.

fbrosson commented:
For 1.0.2f (as well as for the older releases) we have:
* libcrypto.so -> libcrypto.so.1.0.0
* libcrypto.so.1.0.0
* libssl.so -> libssl.so.1.0.0
* libssl.so.1.0.0

So I reverted some of the changes in 1.0.2f and fixed 1.0.0t.
For the `PROVIDES` of 1.0.1h, should I keep the
`        openssl$secondaryArchSuffix = $portVersion compat >= 1.0.0`
Or should I revert it to
`        openssl$secondaryArchSuffix = $portVersion compat >= 1.0.1`

fbrosson commented:
FYI, all 3 recipes build OK on my Haiku ``x86_gcc hrev50072``, although it was a little bit tricky to convince haikuporter to accept to build "old" releases: I had to hide the recipe of 1.1.0~git to build 1.0.2f, then also hide that one to build 1.0.1r, and then also hide that one to build 1.0.0t. 
Haikuported created the hpkg archives (including the _devel) for each release but for some strange reason the hpkgd of 1.0.0t were not copied from ``haikuports/dev-libs/openssl/work-1.0.0t/hpkgs/`` to `haikuports/packages/`..
BTW, an option for haikuporter to prevent it from deleteing outdated hpkgs in `haikuports/packages/` (when it finds out that a recipe no longuer exists) would be welcome.

pulkomandy commented:
For building old version of recipes, see https://github.com/haikuports/haikuporter/issues/26.

fbrosson commented:
Thanks for the pointer. I'll try to see if this issue is fixed in the latest version of haikuporter.
For the record, the version of haikuporter I have (in Haiku x86_gcc2 hrev50072) is 1.0.0-1 (r2429).

korli commented on dev-libs/openssl/openssl-1.0.1r.recipe:
```diff
@@ -11,26 +10,23 @@ OpenSSL is based on the excellent SSLeay library developed by Eric A. Young \
 
-PATCHES="openssl-1.0.1h.patchset"
+ARCHITECTURES="x86_gcc2 x86 ?x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86_gcc2 ?x86 ?x86_64 ?arm ?ppc"
```
Please don't add nonexistent secondary architectures.

fbrosson replied:
ok. I've just remove x86_64 as secondary architectures. Should I also remove arm and ppc (as secondary architectures) ?

korli replied:
The question really is: why were they added in the first place?

fbrosson replied:
Maybe the build creates a dummy self signed certificate, and on cross builds it needs a native build. This is just a guess because I haven't digged the question.

korli commented on dev-libs/openssl/openssl-1.0.1r.recipe:
```diff
@@ -11,26 +10,23 @@ OpenSSL is based on the excellent SSLeay library developed by Eric A. Young \
-	openssl$secondaryArchSuffix = $portVersion compat >= 1.0.1
-	lib:libcrypto$secondaryArchSuffix = 1.0.0 compat >= 1.0
-	lib:libssl$secondaryArchSuffix = 1.0.0 compat >= 1.0
+	openssl$secondaryArchSuffix = $portVersion compat >= 1.0.0
```
compat strings with versions like 1.0.1r don't work right anyway. Maybe we should just drop the compat strings on the package name.

pulkomandy replied:
A better fix would be to use a string that works. IOW, the port version could be 1.0.0.r and then compat 1.0.0 should work.

fbrosson replied:
@pulkomandy, I guess we would need to rename the recipe to openssl-1.0.0.r.recipe in order to have 1.0.0.r in portVersion. Should I do so? Isn't it risky? For the moment I just applied @korli 's suggestion.

korli replied:
FWIW no recipe should depend on openssl packages directly, hence the compat string is useless.

korli commented on dev-libs/openssl/openssl-1.0.2f.recipe:
```diff
@@ -15,13 +15,13 @@ HOMEPAGE="https://www.openssl.org/"
 
 ARCHITECTURES="x86_gcc2 x86 x86_64 ?arm ?ppc"
-SECONDARY_ARCHITECTURES="x86_gcc2 x86 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86 ?x86_64 ?arm ?ppc"
```
Ditto

fbrosson replied:
Also fixed. BTW, are `x86_gcc2` and `x86` the only archivetures allowed in SECONDARY_ARCHITECTURES?
If the answer is yes then I will fix (in a separate PR) all other recipes I have been editing recently.

korli replied:
The answer is yes. Please have a look at other recipes when you do such changes, you won't find any with such secondary architectures.

korli commented on dev-libs/openssl/openssl-1.0.2f.recipe:
```diff
@@ -40,6 +40,11 @@ REQUIRES="
+	REQUIRES="$REQUIRES
+		openssl == $portVersion base
+		"
+fi
```
Could you explain this? I think Adrien removed this as he thought the requirement could be left out.

fbrosson replied:
What I had noticed was that the recipes of `1.0.0t` and `1.0.1h` both had this conditional append for their REQUIRES. So I wrote [in the first comment of this PR](#issuecomment-176677702) that I was not sure wether this change was required/desirable for 1.0.2f or not. From your comment I understand that this conditional append was removed on purpose for the 1.0.2 series. So I just removed it from 1.0.2f. Should I also remove it from 1.0.0t and 1.0.1r ? Or is it safer to just keep that part unchanged?

korli replied:
I don't care about 1.0.0 and 1.0.1 recipes. They will be dropped in a near future.
